cli.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package celvet
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	api "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	apiinstall "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/install"
+	apiv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	structuralschema "k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"k8s.io/apimachinery/pkg/runtime"
+	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+func Lint(args []string, outputWriter io.Writer) (int, error) {
+	if len(args) != 2 {
+		return 1, fmt.Errorf("usage: %s crd-file", args[0])
+	}
+
+	crdFile := args[1]
+	fileBytes, err := ioutil.ReadFile(crdFile)
+	if err != nil {
+		return 1, fmt.Errorf("error reading %s: %s", crdFile, err)
```

Comment:
nit: It's idiomatic to use `%v` or `%w` to format errors.

Reply:
Fixed